src/MOI_wrapper.jl

```
@@ -1623,9 +1614,12 @@ function MOI.get(model::Optimizer, attr::MOI.DualObjectiveValue)
     return MOI.Utilities.get_fallback(model, attr, Float64)
 end
 
-MOI.supports(model::Optimizer, ::MOI.RelativeGap) = model.last_solved_by_mip
-MOI.get(model::Optimizer, ::MOI.RelativeGap) = model.relative_gap
-MOI.supports(::Optimizer, ::MOI.SolveTime) = true
+function MOI.get(model::Optimizer, ::MOI.RelativeGap)
+    if !model.last_solved_by_mip
+        error("RelativeGap is only available for models with integer variables.")
```

I'm not really familiar with this, sorry if that's naive. I was wondering why this throws an error instead of a warning and `return 0.0`.

---

`RelativeGap` is not implemented by Conic and LP solvers so it's ok if GLPK does not implement it when solving an LP.
Moreover, the gap is zero in theory but numerically it might be nonzero.

---

For conic and LPs the user should do `objective_value(model) - dual_objective_value(model)`